tests/e2e/gc_block_resize_retain_policy.go

```diff
@@ -725,6 +741,317 @@ var _ = ginkgo.Describe("[csi-guest] Volume Expansion Tests with reclaimation po
+		svcClient, svNamespace := getSvcClientAndNamespace()
+
+		ginkgo.By("Creating FCD (CNS Volume)")
+		fcdID, err := e2eVSphere.createFCDwithValidProfileID(ctx,
+			"staticfcd"+curtimestring, profileID, diskSizeInMb, defaultDatastore.Reference())
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		ginkgo.By(fmt.Sprintf("Sleeping for %v seconds to allow newly created FCD:%s to sync with pandora",
+			pandoraSyncWaitTime, fcdID))
+		time.Sleep(time.Duration(pandoraSyncWaitTime) * time.Second)
+
+		ginkgo.By("Create CNS register volume with above created FCD")
+		cnsRegisterVolume := getCNSRegisterVolumeSpec(ctx, svNamespace, fcdID, "", svpvcName, v1.ReadWriteOnce)
+		err = createCNSRegisterVolume(ctx, restConfig, cnsRegisterVolume)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		err = waitForCNSRegisterVolumeToGetCreated(ctx,
+			restConfig, svNamespace, cnsRegisterVolume, poll, supervisorClusterOperationsTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		cnsRegisterVolumeName := cnsRegisterVolume.GetName()
+		framework.Logf("CNS register volume name : %s", cnsRegisterVolumeName)
+
+		ginkgo.By("verify created PV, PVC and check the bidirectional reference")
+		svcPVC, err := svcClient.CoreV1().PersistentVolumeClaims(svNamespace).Get(ctx, svpvcName, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		svcPV := getPvFromClaim(svcClient, svNamespace, svpvcName)
+		verifyBidirectionalReferenceOfPVandPVC(ctx, svcClient, svcPVC, svcPV, fcdID)
+
+		//Create PVC,PV  in GC2
+		gcPVC, gcPV := createStaticPVandPVCinGuestCluster(clientNewGc, ctx, namespaceNewGC, svpvcName, diskSize, storageclass, v1.PersistentVolumeReclaimRetain)
+		isGC1PVCCreated := true
+		isGC1PVCreated := true
+
+		defer func() {
+			if isGC1PVCCreated {
+				ginkgo.By("Deleting the gc1 PVC")
+				err = fpv.DeletePersistentVolumeClaim(client, gcPVC.Name, namespaceNewGC)
+			}
+			if isGC1PVCreated {
+				ginkgo.By("Deleting the gc1 PV")
+				err = fpv.DeletePersistentVolume(client, gcPV.Name)
+			}
+
+		}()
+
+		ginkgo.By("Deleting the gc1 PVC")
+		err = fpv.DeletePersistentVolumeClaim(client, gcPVC.Name, namespaceNewGC)
+		isGC1PVCCreated = false
+
+		ginkgo.By("Deleting the gc1 PV")
+		err = fpv.DeletePersistentVolume(client, gcPV.Name)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		err = fpv.WaitForPersistentVolumeDeleted(client, gcPV.Name, poll, pollTimeoutShort)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		isGC1PVCreated = false
+
+		scParameters := make(map[string]string)
+		scParameters[scParamFsType] = ext4FSType
+		scParameters[svStorageClassName] = storagePolicyName
+		storageclassNewGC, err := createStorageClass(client,
+			scParameters, nil, v1.PersistentVolumeReclaimDelete, "", true, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		pvcNew, pvNew, pod, _ := createStaticPVandPVCandPODinGuestCluster(client, ctx, namespace, svpvcName, diskSize, storageclassNewGC, v1.PersistentVolumeReclaimRetain)
```

Comment:
Line exceeding 120 chars - please wrap the line to 120 chars

Reply:
done